### PR TITLE
TMA-618: GoodData Ruby SDK: Allow to work with report folders

### PR DESCRIPTION
**Pull request by vhtien**

### CODE REVIEW

File: lib/gooddata/models/metadata/report.rb [hunk 1]
```diff
@@ -31,6 +31,11 @@ def create(options = { :client => GoodData.connection, :project => GoodData.proj
         fail 'Report needs a title specified' unless title
         summary = options[:summary] || ''
 
+        if options[:folder]
```
Comment (yenkins) [inline lib/gooddata/models/metadata/report.rb]: Calls 'options[:folder]' 2 times (DuplicateMethodCall)

Comment (yenkins) [inline lib/gooddata/models/metadata/report.rb]: Refers to 'options' more than self (maybe move it to another class?) (FeatureEnvy)

File: lib/gooddata/models/metadata/report.rb [hunk 2]
```diff
@@ -51,6 +56,7 @@ def create(options = { :client => GoodData.connection, :project => GoodData.proj
             }
           }
         }
+        report['report']['content']['domains'] << folder if folder
```
Comment (yenkins) [inline lib/gooddata/models/metadata/report.rb]: Calls 'report['report']' 2 times (DuplicateMethodCall)

File: lib/gooddata/models/metadata/report.rb [hunk 3]
```diff
@@ -187,6 +206,16 @@ def purge_report_of_unused_definitions!
       self
     end
 
+    # Add to specified folder
+    #
+    # @param folder [String | GoodData::ReportFolder] folder will contains this report
+    def add_to_folder!(folder)
+      folder_uri = folder.respond_to?(:uri) ? folder.uri : folder
```
Comment (yenkins) [inline lib/gooddata/models/metadata/report.rb]: Manually dispatches method call (ManualDispatch)

File: lib/gooddata/models/metadata/report.rb [hunk 4]
```diff
@@ -31,6 +31,11 @@ def create(options = { :client => GoodData.connection, :project => GoodData.proj
         fail 'Report needs a title specified' unless title
         summary = options[:summary] || ''
 
+        if options[:folder]
+          folder = options[:folder]
+          folder = folder.uri if folder.respond_to?(:uri)
```
Comment (yenkins) [inline lib/gooddata/models/metadata/report.rb]: Manually dispatches method call (ManualDispatch)

File: lib/gooddata/models/metadata/report_folder.rb [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+#
+# Copyright (c) 2010-2017 GoodData Corporation. All rights reserved.
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+require_relative '../metadata'
+require_relative 'metadata'
+
+module GoodData
+  class ReportFolder < GoodData::MdObject
```
Comment (yenkins) [inline lib/gooddata/models/metadata/report_folder.rb]: Has no descriptive comment (IrresponsibleModule)

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-opensource-repository-check/364/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/100/ : SUCCESS in 1m 48s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/410/ : SUCCESS in 1m 44s

**Pull request closed without merging**